src/target_entry.rs

```diff
@@ -0,0 +1,104 @@
+use ffi;
+use glib;
+use glib::translate::*;
+use gobject_ffi;
+use std::mem;
+use TargetFlags;
+
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct TargetEntry {
+    pub target: String,
```

Comment:
There are two other manually-implemented structs in gtk, `RecentData` and `Requisition`, both of which expose all members as public. I'm leaning towards us just doing the same.